bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DecimalType.java

```diff
@@ -165,6 +165,9 @@ public State as(Class<? extends State> target) {
             } else {
                 return UnDefType.UNDEF;
             }
+        } else if (target == HSBType.class) {
+            return new HSBType(DecimalType.ZERO, PercentType.ZERO,
+                    new PercentType(this.toBigDecimal().multiply(new BigDecimal(100))));

```

Comment:
@watou Thanks for that hint, didn't know that. But this valueOf talks about the constructor using a long. Does this argument also fit to ints (I need to look at the code why the static method is better then the constructor.